packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Primitives/CSharpType.cs

```diff
@@ -666,5 +649,24 @@ private CSharpType GetRootType()
 
             return returnType;
         }
+
+        /// <summary>
+        /// Update the instance with given parameters.
+        /// </summary>
+        /// <param name="name">Name of the <see cref="CSharpType"/></param>
+        /// <param name="namespace">Namespace of the <see cref="CSharpType"/></param>
+        public void Update(string? name = null, string? @namespace = null)
+        {
+            if (name != null)
+            {
+                Name = name;
+                FullyQualifiedName = $"{Namespace}.{name}";
```

Comment:
nit: we can just make this property be a lambda getter so we don't have to set it all over the place.